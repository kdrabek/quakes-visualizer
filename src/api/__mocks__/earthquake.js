import Immutable from 'immutable';

export const mockedResponse = {};
const get = jest.fn(async () => Immutable.fromJS(mockedResponse));

export default {
  get,
};
