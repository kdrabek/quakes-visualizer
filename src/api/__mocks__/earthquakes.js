import Immutable from 'immutable';

export const mockedResponse = {
  status: 200,
  data: {
    type: 'FeatureCollection',
    metadata: {
      generated: 1521989257000,
      url: 'https://earthquake.usgs.gov/fdsnws/event/1/query?format=geojson&starttime=2014-01-01&endtime=2014-01-02&minmagnitude=5',
      title: 'USGS Earthquakes',
      status: 200,
      api: '1.5.8',
      count: 2,
    },
    features: [
      {
        type: 'Feature',
        properties: {
          mag: 6.5,
          place: '32km W of Sola, Vanuatu',
          time: 1388592209000,
          updated: 1394151955000,
          tz: 660,
          url: 'https://earthquake.usgs.gov/earthquakes/eventpage/usc000lvb5',
          detail: 'https://earthquake.usgs.gov/fdsnws/event/1/query?eventid=usc000lvb5&format=geojson',
          felt: null,
          cdi: null,
          mmi: 4.84,
          alert: 'green',
          status: 'reviewed',
          tsunami: 1,
          sig: 650,
          net: 'us',
          code: 'c000lvb5',
          ids: ',at00myqcls,pt14001000,usc000lvb5,',
          sources: ',at,pt,us,',
          types: ',cap,geoserve,impact-link,losspager,moment-tensor,moment-tensor,moment-tensor,moment-tensor,nearby-cities,origin,phase-data,shakemap,tectonic-summary,',
          nst: null,
          dmin: 3.997,
          rms: 0.76,
          gap: 14,
          magType: 'mww',
          type: 'earthquake',
          title: 'M 6.5 - 32km W of Sola, Vanuatu',
        },
        geometry: {
          type: 'Point',
          coordinates: [167.249, -13.8633, 187],
        },
        id: 'usc000lvb5',
      },
    ],
    bbox: [120.2389, -13.8633, 10.07, 167.249, 19.0868, 187],
  },
};
const get = jest.fn(async () => Immutable.fromJS(mockedResponse.data));

export default {
  get,
};
