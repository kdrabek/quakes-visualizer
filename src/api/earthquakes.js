import { earthquakeApiUrl } from '../configs';
import { makeRequest } from './base';


const get = (startDate, endDate, minmagnitude) =>
  makeRequest(
    `${earthquakeApiUrl}/query`,
    {
      format: 'geojson',
      starttime: startDate,
      endtime: endDate,
      minmagnitude,
    },
  );

export default {
  get,
};
