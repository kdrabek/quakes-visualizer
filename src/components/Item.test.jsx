import Immutable from 'immutable';
import { shallow } from 'enzyme';
import { mockedResponse } from '../api/__mocks__/earthquakes';
import ListItem, { Item } from './Item';

const feature = mockedResponse.data.features[0];

describe('Item component', () => {
  it('should match snapshot when collapsed', () => {
    const props = {
      properties: Immutable.fromJS(feature.properties),
      id: Immutable.fromJS(feature.id),
      updateMap: () => {},
    };

    const wrapper = shallow(<ListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when expanded', () => {
    const props = {
      properties: Immutable.fromJS(feature.properties),
      id: Immutable.fromJS(feature.id),
      updateMap: () => {},
      classes: { },
    };

    const wrapper = shallow(<Item {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
