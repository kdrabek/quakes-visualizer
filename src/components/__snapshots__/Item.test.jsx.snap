// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should match snapshot when collapsed 1`] = `
<Card
  className="item-card"
  raised={false}
>
  <WithStyles(CardHeader)
    action={
      <WithStyles(IconButton)
        aria-expanded={false}
        className="expand"
        onClick={[Function]}
      >
        <pure(ExpandMore) />
      </WithStyles(IconButton)>
    }
    avatar={
      <WithStyles(Avatar)
        className="magnitude-avatar"
      >
        6.5
      </WithStyles(Avatar)>
    }
    onClick={[Function]}
    subheader="4:03:29 PM, 01/01/2014"
    title="32km W of Sola, Vanuatu"
  />
  <WithStyles(Collapse)
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(CardContent)>
      <div
        className="list-parent"
      >
        <div
          className="list-child"
          key="usc000lvb5-alert"
        >
          <WithStyles(Typography)
            align="center"
          >
            green
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Alert
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-cdi"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Intensity
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-depth"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Depth
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-felt"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Felt Reports
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-tsunami"
        >
          <WithStyles(Typography)
            align="center"
          >
            1
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Tsunami
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-type"
        >
          <WithStyles(Typography)
            align="center"
          >
            earthquake
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Type
          </WithStyles(Typography)>
        </div>
      </div>
      <WithStyles(Typography)
        align="right"
      >
        Report link: 
        <a
          href="https://earthquake.usgs.gov/earthquakes/eventpage/usc000lvb5"
        >
          https://earthquake.usgs.gov/earthquakes/eventpage/usc000lvb5
        </a>
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </WithStyles(Collapse)>
</Card>
`;

exports[`Item component should match snapshot when expanded 1`] = `
<Card
  className="item-card"
  raised={false}
>
  <WithStyles(CardHeader)
    action={
      <WithStyles(IconButton)
        aria-expanded={false}
        className="expand"
        onClick={[Function]}
      >
        <pure(ExpandMore) />
      </WithStyles(IconButton)>
    }
    avatar={
      <WithStyles(Avatar)
        className="magnitude-avatar"
      >
        6.5
      </WithStyles(Avatar)>
    }
    onClick={[Function]}
    subheader="4:03:29 PM, 01/01/2014"
    title="32km W of Sola, Vanuatu"
  />
  <WithStyles(Collapse)
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(CardContent)>
      <div
        className="list-parent"
      >
        <div
          className="list-child"
          key="usc000lvb5-alert"
        >
          <WithStyles(Typography)
            align="center"
          >
            green
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Alert
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-cdi"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Intensity
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-depth"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Depth
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-felt"
        >
          <WithStyles(Typography)
            align="center"
          >
            no info
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Felt Reports
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-tsunami"
        >
          <WithStyles(Typography)
            align="center"
          >
            1
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Tsunami
          </WithStyles(Typography)>
        </div>
        <div
          className="list-child"
          key="usc000lvb5-type"
        >
          <WithStyles(Typography)
            align="center"
          >
            earthquake
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            gutterBottom={true}
            variant="caption"
          >
            Type
          </WithStyles(Typography)>
        </div>
      </div>
      <WithStyles(Typography)
        align="right"
      >
        Report link: 
        <a
          href="https://earthquake.usgs.gov/earthquakes/eventpage/usc000lvb5"
        >
          https://earthquake.usgs.gov/earthquakes/eventpage/usc000lvb5
        </a>
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </WithStyles(Collapse)>
</Card>
`;
